__tests__/app.test.js

```diff
@@ -1,165 +1,204 @@
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("400! Bad request!");
+      });
+  });
+});
+
+describe("GET: 200 - /api/users", () => {
+  test("Responds with an array of objects with user properties", () => {
+    return request(app)
+      .get("/api/users")
+      .expect(200)
+      .then(({ body }) => {
+        const bodyUsers = body.users;
+        expect(bodyUsers).toHaveLength(4);
+        bodyUsers.forEach((user) => {
+          expect(user).toEqual(
+            expect.objectContaining({
+              username: expect.any(String),
+              name: expect.any(String),
+              avatar_url: expect.any(String),
+            })
+          );
+        });
+      });
+  });
+});
+
+describe("PATCH: 200 - /api/articles/:articles_id", () => {
+  test("Updates votes with the required amount and returns the updated article", () => {
+    return request(app)
+      .patch("/api/articles/1")
+      .send({ inc_votes: 1 })
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.article.votes).toBe(101);
+      });
+  });
+  test("400 Error if the article_id is invalid", () => {
+    return request(app)
+      .patch("/api/articles/northcoders")
+      .send({ inc_votes: 1 })
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("400! Bad request!");
+      });
+  });
+  test("400 Error if the inc_votes is an invalid data type", () => {
+    return request(app)
+      .patch("/api/articles/1")
+      .send({ inc_votes: "northcoders" })
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("400! Bad request!");
+      });
+  });
+  test("404 Error if the article_id does not exist", () => {
+    return request(app)
+      .patch("/api/articles/9000")
+      .send({ inc_votes: 1 })
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe("404! this does not exist!");
+      });
+  });
+  test("400 Error if the inc_votes key is not sent with to the body", () => {
+    return request(app)
+      .patch("/api/articles/1")
+      .send()
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("400! Bad request!");
+      });
+  });
+});
+
+describe.only("GET: 200 - /api/articles", () => {
+  test("Return an array of articles with all properties", () => {
+    return request(app)
+      .get("/api/articles")
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.articles).toHaveLength(12);
+        body.articles.forEach((article) => {
+          expect(article).toEqual(
+            expect.objectContaining({
+              author: expect.any(String),
+              title: expect.any(String),
+              topic: expect.any(String),
+              created_at: expect.any(String),
+              article_id: expect.any(Number),
+              votes: expect.any(Number),
+              comment_count: expect.any(Number),
+            })
+          );
+        });
+      });
+  });

```

Comment:
We need to include a tests for when a topic exists, but no articles have been written for it. What would this return?